webpack/assets/javascripts/react_app/components/ForemanModal/ForemanModal.stories.js

```diff
@@ -1,152 +1,231 @@
-import React, { useState } from 'react';
+import React from 'react';
 import { Button } from 'patternfly-react';
 import { storiesOf } from '@storybook/react';
+import storeDecorator from '../../../../../stories/storeDecorator';
 import ForemanModal from '.';
+import { useForemanModal } from './ForemanModalHooks';
 import Story from '../../../../../stories/components/Story';
 
-// This custom Hook is only for the Storybook
-const useModalState = initialModalState => {
-  const [modalOpen, setModalOpen] = useState(initialModalState);
-  const toggleModal = () => setModalOpen(!modalOpen);
-  return [modalOpen, toggleModal];
-};
+storiesOf('Components/ForemanModal', module)
+  .addDecorator(storeDecorator) // add Redux store to story
+  .add('With default header & footer', () =>
+    // using createElement here so that hooks work in stories
+    React.createElement(() => {
+      const { setModalOpen } = useForemanModal({ id: 'default' });
```

Comment:
I'm wondering why do we need the `useForemanModal`, IMO it becomes too complicated to use it.
ATM I need to use `redux` and `useForemanModal` together and it feels like they give me the same functionality (open, close, isOpen). 

It can be implemented differently with `redux` only.

1. on-mount - call the `addModalAction` with the `id`
2. use **only** `redux` actions to close and open
3. use **only** `redux` selectors to get your modal `isOpen` state

It will make my life as a consumer easier, to use the `ForemanModal` I will only need to:
```js
const MyExampleModalUsage = ({
  isMyExampleModalOpen,
  openMyExampleModal,
  closeMyExampleModal,
}) => (
  <div>
    <ForemanModal id="myExampleModal">...</ForemanModal>
    {!isMyExampleModalOpen && (
      <button onClick={openMyExampleModal}>open</button>
    )}
  </div>
);

const MyExampleModalUsageConnected = () => {
  const dispatch = useDispatch();
  return (
    <MyExampleModalUsage
      isMyExampleModalOpen={useSelector(selectIsModalOpen('myExampleModal'))}
      openMyExampleModal={useDispatch(() => setModalOpen('myExampleModal'))}
      closeMyExampleModal={useDispatch(() => setModalClose('myExampleModal'))}
    />
  );
};
```

Reply:
@sharvit `useForemanModal` is totally optional.  It's totally valid to choose just to use plain Redux actions, or action creators, or the Hook.  I think any of those three approaches are fine.  (Also, the hook only works if your component is written using Hooks.  So it naturally can't be the only option.)

Reply:
If we will allow consumers to achieve the same, using 2 different ways, it will be confusing for consumers about what they should choose.
It also means we should maintain 2 different interfaces for the long term.

Reply:
@sharvit If we decide not to provide a Hook just because Hooks might be confusing, then we shouldn't allow the use of React hooks at all in Foreman.

Rather, I think Hooks are the future, and for a new learner classes are actually the more confusing option.  It's just because we don't have an "empty cup" and are used to classes that it's harder for us to see this as more experienced React devs.

As far as the long term, we should expect to write more and more with hooks and less with classes, and during the transition period we will have to live with it.  Just like we live with our Angular and jQuery today. ;) But I think it's worth it!

The developers of React do not deprecate class-based components, and they also offer hooks.  Just as they offer both options, I think it's fine for us to offer parallel options.

Reply:
@sharvit If you use `useForemanModal` then you don't need to use Redux explicitly at all-- the hook uses it for you behind the scenes.  It's reusing the stateful logic (the ForemanModal-related selectors and dispatches) that you otherwise would have to rewrite every time you use ForemanModal.

However, there's always an argument for being more explicit.  So if you want to use `useSelector` and `useDispatch` every time, and not wrap it in `useForemanModal` that's just fine.